src/store/initial-state.js

```diff
@@ -44,24 +44,16 @@ export const mergeLocalStorage = state => {
 };
 
 /**
- * Add custom state overrides from props, URL flags, etc
- * @param {object} state App state
+ * Prepare the pipeline data part of the state. This part is separated so that it
+ * can be reset without overriding user settings.
  * @param {object} props Props passed to App component
  */
-export const overrideInitialState = (state, props) => {
-  // Override flag defaults with URL values
-  state.flags = Object.assign(state.flags, getFlagsFromUrl());
-  // Override theme if set in props
-  if (props.theme) {
-    state.theme = props.theme;
-  }
-  // Override button visibility if set in props
-  if (props.visible) {
-    state.visible = Object.assign(state.visible, props.visible);
-  }
+export const preparePipelineState = props => {
+  // Normalize raw data, and apply saved state from localStorage
+  const state = mergeLocalStorage(normalizeData(props.data));
   // Turn layers off if there are no layers present:
   if (!state.layer.ids.length) {
-    state.visible.layers = false;
+    state.visible = Object.assign({}, state.visible, { layers: false });
```

Comment:
Could be written as `state.visible = { ...state.visible, layers: false }` which I feel shows the intention of creating a new object more clearly, assuming that was the reason for the change?

Reply:
I have a slight preference for Object.assign here (which in practice is more often than not used to create a new object). Also taking transpiled size into account, there's a decent chance that Object.assign will be smaller. But overall I really don't see much significant difference between the two.

Reply:
Yeah it's not a big deal for sure, I know what you mean as I've seen `Object.assign` being used for cloning like this before object spread was a thing, but since we have support for it I feel it's a more natural way to express this.